### PR TITLE
Adding unit tests for helper

### PR DESCRIPTION
**Pull request by moizjv**
- @jlipps @Jonahss please review

### CODE REVIEW

File: test/unit/android-helper-specs.js [hunk 1]
```diff
@@ -39,7 +44,233 @@ describe('Android Helpers', () => {
...
+      mocks.adb.verify();
+    });
+    it('should return when appPackage & appActivity are already present',
+      async () => {
+        mocks.adb.expects('packageAndLaunchActivityFromManifest').never();
+        await helpers.getLaunchInfoFromManifest(adb, {app: "foo", appPackage: "bar",
+          appActivity: "act"});
+        mocks.adb.verify();
+    });
+    it('should return package and launch activity from moanifest', async () => {
```
Comment (Jonahss) [inline test/unit/android-helper-specs.js]: typo: manifest

File: test/unit/android-helper-specs.js [hunk 2]
```diff
@@ -39,7 +44,233 @@ describe('Android Helpers', () => {
...
+      await helpers.resetApp(adb, localApkPath, pkg, false);
+      mocks.adb.verify();
+      mocks.fs.verify();
+      mocks.helpers.verify();
+    });
+  }));
 
-
-
+  describe.skip('reinstallRemoteApk', withMocks({adb}, (mocks) => {
```
Comment (Jonahss) [inline test/unit/android-helper-specs.js]: `.skip`?

File: test/unit/android-helper-specs.js [hunk 3]
```diff
@@ -39,7 +44,233 @@ describe('Android Helpers', () => {
...
+      mocks.helpers.expects('getRemoteApkPath').returns(false);
+      mocks.adb.expects('fileExists').returns(true);
+      mocks.adb.expects('isAppInstalled').returns(true);
+      mocks.helpers.expects('resetApp').once().returns("");
+      await helpers.installApkRemotely(adb, localApkPath, pkg, true);
+      mocks.adb.verify();
+      mocks.fs.verify();
+      mocks.helpers.verify();
+    });
+    it.skip('should push and reinstall apk when apk is not installed', async () => {
```
Comment (Jonahss) [inline test/unit/android-helper-specs.js]: `skip`?

File: lib/android-helpers.js [hunk 4]
```diff
@@ -81,12 +81,13 @@ helpers.getActiveDevice = async function (adb, udid) {
...
     }
     deviceId = udid;
+    emPort = adb.getPortFromEmulatorString(deviceId);
   } else {
     deviceId = devices[0].udid;
     emPort = adb.getPortFromEmulatorString(deviceId);
   }
   logger.info(`Found device: ${deviceId}`);
-  return [deviceId, emPort];
+  return {deviceId, emPort};
```
Comment (jlipps) [inline lib/android-helpers.js]: let's not forget to change appium-selendroid after we merge this

File: lib/android-helpers.js [hunk 5]
```diff
@@ -118,7 +119,7 @@ helpers.getLaunchInfoFromManifest = async function (adb, opts) {
   return {appPackage, appWaitPackage, appActivity, appWaitActivity};
 };
 
-helpers.getRemoteApkPath = async function (localApkMd5) {
+helpers.getRemoteApkPath = function (localApkMd5) {
```
Comment (jlipps) [inline lib/android-helpers.js]: :+1:

File: lib/android-helpers.js [hunk 6]
```diff
@@ -131,9 +132,8 @@ helpers.resetApp = async function (adb, localApkPath, pkg, fastReset) {
   } else {
     logger.debug("Running old fashion reset (reinstall)");
     let apkMd5 = await fs.md5(localApkPath);
-    let remotePath = await helpers.getRemoteApkPath(apkMd5, localApkPath);
-    if (await adb.fileExists(remotePath)) {
-    } else {
+    let remotePath = helpers.getRemoteApkPath(apkMd5, localApkPath);
+    if (!await adb.fileExists(remotePath)) {
```
Comment (jlipps) [inline lib/android-helpers.js]: i believe you need parens around the `await` phrase?

File: lib/android-helpers.js [hunk 7]
```diff
@@ -256,7 +257,7 @@ helpers.unlock = async function (adb) {
       category: "android.intent.category.LAUNCHER",
       flags: "0x10200000"
     });
-    if (await adb.screenIsLocked()) {
+    if (!await adb.isScreenLocked()) {
```
Comment (jlipps) [inline lib/android-helpers.js]: good catch, but same thing about parens

File: test/unit/android-helper-specs.js (hunk 1 above)
Comment (jlipps) [inline test/unit/android-helper-specs.js]: lol moanifest. more accurate.

File: lib/android-helpers.js (hunk 6 above)
Comment (Jonahss) [inline lib/android-helpers.js]: I think the syntax there is ok. isn't it?

Comment (Jonahss) [inline lib/android-helpers.js]: tested, `!` operator is applies to the awaited result of an await statement.

Comment (jlipps) [inline lib/android-helpers.js]: \o/

**Pull request closed without merging**